src/dinit-iostream.cc

```diff
@@ -0,0 +1,759 @@
+#include <algorithm>
+#include <limits>
+#include <cstring>
+#include <fcntl.h>
+
+#include <dinit-iostream.h>
+#include <baseproc-sys.h>
+
+// See dinit-iostream.h for inforamtion about the usage
+
+namespace {
+
+// A simple trick for calculating how much digits can be in a type with compile-time
+// integer logarithm like function.
+template <typename T> constexpr unsigned TYPE_MAX_NUMDIGITS(T num = std::numeric_limits<T>::max(),
+        unsigned pow = 0)
+{
+    return (num == 0) ? pow : TYPE_MAX_NUMDIGITS(num / 10, pow + 1);
+}
+
+constexpr unsigned INT_NUMDIGITS = TYPE_MAX_NUMDIGITS<int>() + 1;
+constexpr unsigned UINT_NUMDIGITS = TYPE_MAX_NUMDIGITS<unsigned>() + 1;
+constexpr unsigned LONG_NUMDIGITS = TYPE_MAX_NUMDIGITS<long>() + 1;
+constexpr unsigned ULONG_NUMDIGITS = TYPE_MAX_NUMDIGITS<unsigned long>() + 1;
+
+} /* anonymous namespace */
+
+namespace dio {
+
+// class io_base
+streambuf *io_base::get_buf() noexcept
+{
+    return buf.get();
+}
+
+bool io_base::is_open() noexcept
+{
+    return (fd >= 0);
+}
+
+void io_base::set_fd(const int newfd) noexcept
+{
+    fd = newfd;
+}
+
+int io_base::get_fd() noexcept
+{
+    return fd;
+}
+
+// class ostream
+ssize_t ostream::put(const char *msg, size_t count) noexcept
+{
+    if (!good()) {
+        return -1;
+    }
+
+    // If the current buffer space is not enough to store the message,
+    // we fill the buffer as much as possible and subtract the number of
+    // characters used to fill the buffer from the "count" and add it
+    // to this variable.
```

Comment:
These comment lines (here and in many other places) are wrapping way too short, the convention is to wrap just before 100 characters (up to maximum of 110 if wrapping before 100 is awkward).

Reply:
Hmm, I thought it's 80 characters.

Reply:
It is not completely consistent throughout the current code base, but 100/110 is the documented standard for new code, see `CODE-STYLE`